.github/workflows/phpunit-tests.yml

```diff
@@ -0,0 +1,128 @@
+                        '7.1',
+                        '7.2',
+                        '7.3',
+                        '7.4',
+                        '8.0',
+                        '8.1',
+                        '8.2',
+                    ]
+                os: [ubuntu-latest]
+                multisite: [false, true]
+
+        env:
+            WP_ENV_PHP_VERSION: ${{ matrix.php }}
+
+        steps:
+            - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+
+            - name: Set up Node.js
+              uses: actions/setup-node@8c91899e586c5b171469028077307d293428b516 # v3.5.1
+              with:
+                  node-version-file: '.nvmrc'
+                  cache: npm
+
+            ##
+            # This allows Composer dependencies to be installed using a single step.
+            #
+            # Since the tests are currently run within the Docker containers where the PHP version varies,
+            # the same PHP version needs to be configured for the action runner machine so that the correct
+            # dependency versions are installed and cached.
+            ##
+            - name: Set up PHP
+              uses: shivammathur/setup-php@1a18b2267f80291a81ca1d33e7c851fe09e7dfc4 # v2.22.0
+              with:
+                  php-version: '${{ matrix.php }}'
+                  coverage: none
+
+            # Ensure that Composer installs the correct versions of packages.
+            - name: Override PHP version in composer.json
+              run: composer config platform.php ${{ matrix.php }}
+
+            # The spatie/phpunit-watcher package is not compatible with PHP < 7.2.
+            # It must be removed before running the tests.
+            - name: Remove incompatible Composer packages
+              if: ${{ matrix.php < '7.2' }}
+              run: composer remove spatie/phpunit-watcher --dev --no-update
+
+            # Since Composer dependencies are installed using `composer update` and no lock file is in version control,
+            # passing a custom cache suffix ensures that the cache is flushed at least once per week.
+            - name: Install Composer dependencies
+              uses: ramsey/composer-install@83af392bf5f031813d25e6fe4cd626cdba9a2df6 # v2.2.0
+              with:
+                  custom-cache-suffix: $(/bin/date -u --date='last Mon' "+%F")
+
+            - name: Install npm dependencies
+              run: |
+                  npm ci
+                  npm run build
+
+            - name: Docker debug information
+              run: |
+                  docker -v
+                  docker-compose -v
+
+            - name: General debug information
+              run: |
+                  php --version
+                  npm --version
+                  node --version
+                  curl --version
+                  git --version
+                  svn --version
+                  composer --version
+                  locale -a
```

Comment:
Is all this necessary or mostly useful while developing this branch? Same question for the Docker debug info step below.

Reply:
It's not necessary, but it's worthwhile.
E.g., it helps to confirm which version of Composer or PHP is used when installing Composer packages.

Reply:
I'm asking because we don't currently do this for our other workflows, and it's a fair amount of noise, and some duplication (e.g. PHP and composer versions are already available through the matrix and through the PHP install step).

It's not problematic, per se, just curious why it's here as it seems like a step (to remove this debug logging) that might have been overlooked before prepping for final merge into `trunk`.

Reply:
>  because we don't currently do this for our other workflows

We actually [do this](https://github.com/WordPress/wordpress-develop/blob/trunk/.github/workflows/phpunit-tests.yml#L138).

> PHP and composer versions are already available through the matrix and through the PHP install step

Yes, they are through the matrix, but how do we know that these PHP versions are actually being used?
Also, in the matrix, we only specify major and minor PHP versions (the same way it's implemented in Core).
The logging allows us to find the exact PHP version (including the point release) used when running that particular CI job.

Reply:
> to know the exact PHP version (including the point release) used when running particular CI jobs.

That information is already available via the `setup-php` step.

Jonathan and me actually discussed removing a lot of this kind of debug noise from the WP Core workflows recently as the information is already available within the workflow.

Reply:
> That information is already available via the setup-php step.

I've removed logging Composer and PHP versions.
cb3ee089ee66b68af28305833e5868f7d517454d
2dfb8452e972553650ac309e497cb0fee252be15